fix(eks): creating a `ServiceAccount` in a different stack than the `Cluster` creates circular dependency between the two stacks (#9701)
aws/aws-cdk: packages/@aws-cdk/aws-eks/README.md

The `KubernetesManifest` construct or `cluster.addManifest` method can be used
to apply Kubernetes resource manifests to this cluster.

> > When using `cluster.addManifest`, the manifest is created inside the cluster stack. If the manifest contains attributes from a different stack that depends on the cluster stack, a circular dependency will be created and you will get a synth time error. To avoid this, directly use `new KubernetesManifest` to create the manifest in the other stack.
> 

Reviewer: ```suggestion
> When using `cluster.addManifest`, the manifest construct is defined within the cluster's stack scope. If the manifest contains attributes from a different stack which depend on the cluster stack, a circular dependency will be created and you will get a synth time error. To avoid this, directly use `new KubernetesManifest` to create the manifest in the scope of the other stack.

```

Author: 👍

The `HelmChart` construct or `cluster.addChart` method can be used
to add Kubernetes resources to this cluster using Helm.

> > When using `cluster.addChart`, the chart is created inside the cluster stack. If the chart contains attributes from a different stack that depends on the cluster stack, a circular dependency will be created and you will get a synth time error. To avoid this, directly use `new HelmChart` to create the chart in the other stack.

Reviewer: Same fixes as above

Author: 👍
